nominations/models.py

```diff
@@ -355,6 +470,76 @@ def save(self, *args, **kwargs):
         if self.vol_notes and not self.staff_notes:
             self.staff_notes = self.vol_notes
 
+    def create_related_objects(self):
+        """Create related nomination and questionnaire for application."""
+        if not self.nomination:
+            self.nomination = Nomination.objects.create()
+
+        if not self.questionnaire:
+            self.questionnaire = Questionnaire.objects.create()
+
+    def generate_application_status(self):
+        """Returns a generated status based on completion of various items."""
+
+        if self.nomination.status == 'incomplete':
+            if self.questionnaire.status == 'complete':
+                status = 'needs-group-form'
+            else:
+                status = 'needs-group-form-and-questionnaire'
+        else:
+            # nomination complete
+            if self.questionnaire.status == 'incomplete':
+                # needs questionaire
+                status = 'needs-questionnaire'
+            else:
+                # questionnaire complete
+                status = 'incomplete'
+        return status
+
+    def is_editable(self):
+        """Returns whether the user can edit this application."""
+
+        # TODO possibly move this to settings.py or env
+        editable_statuses = [
+            'needs-group-form-and-questionnaire',
+            'needs-questionnaire',
+            'needs-group-form',
+            'incomplete'
+        ]
+
+        if self.status in editable_statuses:
+            return True
+        else:
+            return False
+
+    def __unicode__(self):
+        return str(self.group) + ' - ' + self.candidate_first_name + ' ' + self.candidate_last_name
+
+    def save(self, *args, **kwargs):
+        if not self.nomination or not self.questionnaire:
+            self.create_related_objects()
+
+        self.auto_populate_research_fields()
+
+        # TODO move to settings.py or env?
+        # dont run status generator if set to these
+        exempt_statuses = [
+            'submitted',
+            'needs-research',
+            'needs-staff-review',
+            'under-review',
+            'approved',
+            'removed',
+            'expired',
+            'hold'
+        ]
+
+        if self.status not in exempt_statuses:
+            self.status = self.generate_application_status()
```

Comment:
I suggest moving this "exempt" logic to `generate_application_status` method. Then this part here can be rewritten like
```
status_new = self.generate_application_status()
if self.status != status_new:
            self.status = status_new
```
or simply
```
self.status = self.generate_application_status()
```